### ISSUE TITLE
strange timeout of 2 minutes when sending mail

### ISSUE THREAD

**Issue by fansari** [flathub/org.gnome.Evolution#96]
I have tested flatpak version 3.44.4 on Fedora Silverblue 36.

Unfortunately I cannot send mails. When sending it is just hanging. I don't see anything incoming on my mail container.

![Screenshot from 2022-09-10 15-17-01](https://user-images.githubusercontent.com/7888607/189486003-e79574c9-921e-4f26-a903-d0e585c7cf74.png)

On the command line (flatpak run org.gnome.Evolution) I see this - but this message appears right on startup.

```
(evolution.bin:49): e-mail-engine-WARNING **: 15:34:30.667: Failed to add service 'Sendmail' (sendmail): No provider available for protocol “sendmail”
```
My sending configuration is SMTP to port 587 to  a local container.

The same configuration works with the RPM version.

Additional info: I had used the fedora remote before and this worked but there was another issue (empty canvas when opening a mail). This is why I want to test the version from the flathub remote.

https://bugzilla.redhat.com/show_bug.cgi?id=2124062

The answer from RedHat is that the flatpak version 3.44.4 is not applicable for Fedora 36.

Therefore I have uninstalled the flatpak version and reinstalled the RPM.

**Comment by fansari** [comment]
I have updated to Fedora 37 Sliverblue and tried version 3.46.0.

It is still hanging when sending mails. Nothing happens. What can be the reason for this???

**Comment by Sesivany** [comment]
Did it work with flatpak before? I think the problem is that it uses sendmail, but sendmail has not been included in the flatpak. We decided not to include it because the feature is used by a fairly small group of users. But we could revisit that, I could try sendmail and see how it works in the flatpak.

**Comment by fansari** [comment]
It worked before. I cannot remember when exactly the problem started. I used an older version from the fedora repo. This had another issue (I always got an emtpy canvas when opening mail and only after several attempts it finally opened). Then I changed to the version of the flathub repo.

But I am unsure whether the issue really started then. Maybe I just did not send mail right away and only saw that mails open directly without this empty canvas issue.

I askid the question about sendmail before - but I remember you told me that sendmail is not supported. But as it turned sendmail was not necessary to deliver mails to a local container since I simply can set my container as SNMP serer.

But now even this does not work anymore correctly. There is this strange timeout affer the connect and I could not figure out in any logfile what happens at this point.

**Comment by mcrha** [comment]
Before sending `EHLO`, the SMTP code tries to figure out local address/FQDN, to know what to send to the server instead of the IP address within the `EHLO`, command, thus I guess it's waiting for a response from some server/DNS until it timeouts after two minutes and  only then it continues.

I also noticed mixed IPv6 and IPv4 addresses here. I saw a user to whom the IPv6 didn't work properly recently, causing similar timeouts on various places. Disabling IPv6 helped.

**Comment by mcrha** [comment]
Hard to tell why the sandbox/flatpak resolves routes differently than the RPM version. It might be something with the Flatpak way, I do not know.

**Comment by fansari** [comment]
Today again it took quite long to send a mail.

```
Oct 11 16:25:17 mail postfix/smtpd[73]: connect from host.containers.internal[10.90.0.1]
Oct 11 16:27:27 mail postfix/smtpd[73]: 50D95252481F1: client=host.containers.internal[10.90.0.1]
```

I have now set my DNS of 10.90.0.3 to mail.containers.internal and after this it worked.

But I will have to verify whether this runs stable now.

**Comment by fansari** [comment]
This seems to have fixed it. I sent mails today and it worked without this delay.

I have put both host.containers.internal and mail.containers.internal into my DNS.
